test/integration/targets/tower_workflow_template/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+- name: Create an SCM Credential
+  tower_credential:
+    name: SCM Credential for JT
+    organization: Default
+    kind: scm
+
+- name: Create a Demo Project
+  tower_project:
+    name: Job Template Test Project
+    organization: Default
+    state: present
+    scm_type: git
+    scm_url: https://github.com/ansible/ansible-tower-samples.git
+    scm_credential: SCM Credential for JT
+  register: result
+
+- name: Update the project (to clone the git repo)
+  uri:
+    url: "https://{{ lookup('env', 'TOWER_HOST') }}/api/v2/projects/{{ result.id }}/update/"
+    method: POST
+    user: "{{ lookup('env', 'TOWER_USERNAME') }}"
+    password: "{{ lookup('env', 'TOWER_PASSWORD') }}"
+    validate_certs: false
+    status_code: 202
+    force_basic_auth: true
+
+- name: Wait for the project to be status=successful
+  uri:
+    url: "https://{{ lookup('env', 'TOWER_HOST') }}/api/v2/projects/{{ result.id }}/"
+    method: GET
+    user: "{{ lookup('env', 'TOWER_USERNAME') }}"
+    password: "{{ lookup('env', 'TOWER_PASSWORD') }}"
+    validate_certs: false
+    force_basic_auth: true
+    return_content: true
+  register: result
+  until: result.json.status == "successful"
+  retries: 60
+  delay: 1
+
+- name: Create a Job Template
+  tower_job_template:
+    name: my-job-1
+    project: Job Template Test Project
+    inventory: Demo Inventory
+    playbook: hello_world.yml
+    machine_credential: Demo Credential
+    job_type: run
+    state: present
+
+- name: Create a second Job Template
+  tower_job_template:
+    name: my-job-2
+    project: Job Template Test Project
+    inventory: Demo Inventory
+    playbook: hello_world.yml
+    machine_credential: Demo Credential
+    job_type: run
+    state: present
+
+- name: Create a workflow job template
+  tower_workflow_template:
+    name: my-workflow
+    schema: '[{"success": [{"job_template": "my-job-1"}], "job_template": "my-job-2"}]'
```

Comment:
Actually, I'm a little skeptical that this worked correctly. I can't easily recognize the pattern here, because the root-level keys should be resources like job templates, not either a dictionary of "success". Also, this is a string... which makes me wonder why Ansible accepted it when the arg spec gave dict.

Reply:
`'[{"success": [{"job_template": "my-job-1"}], "job_template": "my-job-2"}]'` creates a dummy workflow were the two jobs are linked to the starting point (that's how it appears in my AWX UI). I thought it was enough to test the workflow creation. I can write something more complex if needed.

Regarding Ansible accepting that JSON as a string, I guess it's the normal behaviour as I'm not requesting any `type=` in my AnsibleModule.argument_spec call. Maybe I should specify `type='str'` in the code for it to be clearer but as it was done without it in many modules I checked, I did the same.

Reply:
I believe you, that probably does work, it's just that I got hung up on the syntax.

Consider, for your description

> dummy workflow were the two jobs are linked to the starting point

I would expect just

```yaml
    schema: '["job_template": "my-job-1", "job_template": "my-job-2"]'
```

The root-level node doesn't need the success/failure distinction (they're root level, they always run).

Granted, I see that this is not a dictionary, it's a list. Nonetheless, it's still accepted by the python JSON parser.

having both "success" and "job_template" inside of the same dictionary strikes me as a little bit of type confusion. I'm not surprised that tower-cli gobbles that up without complaining, but maybe it does not do much (if any) validation, so maybe it should error in that case.